backend/internal/application/error_constants.go

```diff
@@ -129,70 +128,138 @@ var (
 	ErrorCertificateClientError = serviceerror.ServiceError{
 		Type:             serviceerror.ClientErrorType,
-		Code:             "APP-1021",
+		Code:             "APP-1022",
 		Error:            "Certificate operation failed",
 		ErrorDescription: "An error occurred while processing the application certificate",
 	}
 	// ErrorApplicationAlreadyExistsWithName is the error returned when an application with the same name
 	// already exists.
 	ErrorApplicationAlreadyExistsWithName = serviceerror.ServiceError{
 		Type:             serviceerror.ClientErrorType,
-		Code:             "APP-1022",
+		Code:             "APP-1023",
 		Error:            "Application already exists",
 		ErrorDescription: "An application with the same name already exists",
 	}
 	// ErrorApplicationAlreadyExistsWithClientID is the error returned when an application with the same client ID
 	// already exists.
 	ErrorApplicationAlreadyExistsWithClientID = serviceerror.ServiceError{
 		Type:             serviceerror.ClientErrorType,
-		Code:             "APP-1023",
+		Code:             "APP-1024",
 		Error:            "Application with client ID already exists",
 		ErrorDescription: "An application with the same client ID already exists",
 	}
-	// ErrorPublicClientInvalidConfiguration is the error returned when a public client has invalid configuration.
-	ErrorPublicClientInvalidConfiguration = serviceerror.ServiceError{
+	// ErrorJWKSConfigurationConflict is the error returned when both jwks and jwks_uri are provided.
+	ErrorJWKSConfigurationConflict = serviceerror.ServiceError{
 		Type:             serviceerror.ClientErrorType,
-		Code:             "APP-1024",
-		Error:            "Invalid configurations for public client",
-		ErrorDescription: "One or more configurations specified are not valid for the public clients",
+		Code:             "APP-1025",
+		Error:            "Invalid JWKS configuration",
+		ErrorDescription: "Cannot specify both 'jwks' and 'jwks_uri' parameters",
+	}
+	// ErrorJWKSUriNotHTTPS is the error returned when jwks_uri does not use HTTPS.
+	ErrorJWKSUriNotHTTPS = serviceerror.ServiceError{
+		Type:             serviceerror.ClientErrorType,
+		Code:             "APP-1026",
+		Error:            "Invalid JWKS URI scheme",
+		ErrorDescription: "'jwks_uri' must use HTTPS scheme",
+	}
+	// ErrorPublicClientInvalidAuthMethod is the error returned when public client uses invalid auth method.
+	ErrorPublicClientInvalidAuthMethod = serviceerror.ServiceError{
+		Type:             serviceerror.ClientErrorType,
+		Code:             "APP-1027",
+		Error:            "Invalid public client configuration",
+		ErrorDescription: "Public clients must use only 'none' as token endpoint authentication method",
+	}
+	// ErrorPublicClientInvalidGrantType is the error returned when public client uses invalid grant type.
+	ErrorPublicClientInvalidGrantType = serviceerror.ServiceError{
```

Comment:
Alternatively what if we define a single error constant/ code for "Invalid public client configuration" errors and customize the description with `serviceerror.CustomServiceError()`?